hack/run_test

```diff
@@ -348,11 +350,51 @@ function setup_replication_cluster() {
   test_slave_visibility $cid_suffix
 }
 
+function run_master_restart_test() {
+  echo "Testing failed master restart"
+  local cluster_args="-e POSTGRESQL_ADMIN_PASSWORD=pass -e POSTGRESQL_MASTER_USER=$PGUSER -e POSTGRESQL_MASTER_PASSWORD=$PASS"
+  local cid_suffix="mrestart"
+  local table_name="t1"
+  # Global variable for setup_replication_cluster and test_value_replication
+  max_attempts=30
+
+  local volume_dir
+  volume_dir=`mktemp -d --tmpdir pg-testdata.XXXXX`
+  chmod a+rwx ${volume_dir}
+  master_args="-v ${volume_dir}:/var/lib/pgsql/data:Z"
+
+  # Setup the cluster
+  local slave_num=2
+  setup_replication_cluster $slave_num
+
+  echo "Kill the master and create a new one"
+  local cidfile=$CIDFILE_DIR/master-$cid_suffix.cid
+  docker kill $(cat $cidfile)
+  # Don't forget to remove its .cid file
+  rm $cidfile
+
+  run_master $cid_suffix
+  CONTAINER_IP=$(get_container_ip master-$cid_suffix.cid)
+
+  # Check if the new master sees existing slaves
+  test_slave_visibility $cid_suffix
+
+  # Check if the replication works
+  test_value_replication $table_name
+
+  # Cleanup the volume
+  docker stop $(cat $cidfile)
+  local rm_cmd='/bin/rm -rf /var/lib/pgsql/data/*'
+  docker run $master_args --rm $IMAGE_NAME /bin/sh -c "$rm_cmd"
+  rmdir ${volume_dir}
+}
+
 function run_replication_test() {
   echo "Testing master-slave replication"
   local cluster_args="-e POSTGRESQL_ADMIN_PASSWORD=pass -e POSTGRESQL_MASTER_USER=$PGUSER -e POSTGRESQL_MASTER_PASSWORD=$PASS"
   local cid_suffix="basic"
   local table_name="t1"
+  # Global variable for setup_replication_cluster and test_value_replication
```

Comment:
With a setting such as this that is used in polling loops and probably won't be changed much from its default value, I felt like it would be best to use it as a global variable. Although with the addition of another test I am now setting it in two places so I will fix that.